tools/run_tests/python_utils/upload_rbe_results.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Uploads RBE results to BigQuery"""
+
+import argparse
+import os
+import json
+import sys
+import urllib2
+import uuid
+
+gcp_utils_dir = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '../../gcp/utils'))
+sys.path.append(gcp_utils_dir)
+import big_query_utils
+
+_DATASET_ID = 'jenkins_test_results'
+_DESCRIPTION = 'Test results from master RBE builds on Kokoro'
+# 90 days in milliseconds
+_EXPIRATION_MS = 90 * 24 * 60 * 60 * 1000
+_PARTITION_TYPE = 'DAY'
+_PROJECT_ID = 'grpc-testing'
+_RESULTS_SCHEMA = [
+    ('job_name', 'STRING', 'Name of Kokoro job'),
+    ('build_id', 'INTEGER', 'Build ID of Kokoro job'),
+    ('build_url', 'STRING', 'URL of Kokoro build'),
+    ('test_target', 'STRING', 'Bazel target path'),
+    ('result', 'STRING', 'Test or build result'),
```

Comment:
This will be a string of 'PASSED', 'BUILT', 'UNKNOWN', and 'FAILED'. Right now targets include both tests and libraries/binaries built. If we remove the latter, we probably won't have the 'BUILT' status anymore.

`build_url` contains the Sponge link in the form (https://sponge.corp.google.com/invocation?id=d469bd1e-1e41-4bb7-bc5e-10e6b14f4500). I'm open to switching this to ResultStore instead, if that makes deflaking efforts easier.

Reply:
We probably need to add 'CACHED'.

Reply:
All the test actions that I'm querying return `CACHE_MISSED`, and IIRC, there was an issue with ResultStore propagating cache hits as `PASSED`. I'll look deeper into whether we can get the correct cache hit results.